flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/extraction/DataTypeExtractor.java

```diff
@@ -595,4 +605,36 @@ private DataType closestBridging(DataType dataType, @Nullable Class<?> clazz) {
 		}
 		return dataType;
 	}
+
+	/**
+	 * Data type hints are allowed on top of {@link DataView}s. They are validated and mapped to the underlying
+	 * collection by this method.
+	 */
+	private DataType handleDataViewHints(DataType dataType, @Nullable Class<?> clazz) {
+		if (clazz == null || !DataView.class.isAssignableFrom(clazz)) {
+			return dataType;
+		}
+
+		// data type went through regular extraction logic
+		if (isCompositeType(dataType.getLogicalType())) {
+			return dataType;
```

Comment:
out of curiosity: Is it possible to end up here?

Reply:
A regular
```
class Acc {
 public MapView<String, Long> view;
}
```
should end up there. But I will verify again.

Reply:
This code path is tested by:
```
			TestSpec
				.forType(
					"Data view with default extraction",
					AccumulatorWithDefaultDataView.class)
				.lookupExpects(Object.class)
				.expectDataType(
					DataTypes.STRUCTURED(
						AccumulatorWithDefaultDataView.class,
						DataTypes.FIELD(
							"listView",
							ListView.newListViewDataType(DataTypes.RAW(new GenericTypeInfo<>(Object.class))))
					)
				)
```